setup.py

```diff
@@ -1,73 +1,102 @@
+import tarfile
+import tempfile
+import zipfile
 from distutils.command.build import build as orig_build
 from distutils.core import Command
-from hashlib import sha512
 from http import HTTPStatus
 from urllib.request import urlopen
 
 from setuptools import setup
 from setuptools.command.install import install as orig_install
 
-SHELLCHECK_VERSION = '0.7.0'
+SHELLCHECK_VERSION = '0.7.1'
 PY_VERSION = '1'
 
 
 def get_arch() -> str:
     if sys.platform.startswith('linux'):
-        # TODO(rhee): detect "linux-aarch64" and "linux-armv6hf"
-        return 'linux-x86_64'
+        # TODO(rhee): detect "linux.aarch64" and "linux.armv6hf"
+        return 'linux.x86_64.tar.xz'
     elif sys.platform == 'darwin':
-        return 'darwin-x86_64'
+        return 'darwin.x86_64.tar.xz'
     elif sys.platform == 'win32':
-        return 'exe'
+        return 'zip'
     else:
         raise ValueError('Unsupported platform')
 
 
-def get_executable_name() -> str:
+def get_artifact_name() -> str:
     return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
+    return (
+        'https://github.com/koalaman/shellcheck/releases/download/'
+        f'v{SHELLCHECK_VERSION}/{get_artifact_name()}'
+    )
 
 
 def download(url: str) -> bytes:
-    with urlopen(url) as resp, urlopen(url + '.sha512sum') as shasum_resp:
+    with urlopen(url) as resp:
         code = resp.getcode()
-        scode = shasum_resp.getcode()
-        if code != HTTPStatus.OK or scode != HTTPStatus.OK:
-            raise ValueError(f'HTTP failure. Codes: {code}, {scode}')
-
-        data = resp.read()
-        actual_shasum = sha512(data).hexdigest()
-        expected_shasum = shasum_resp.read().decode('utf8').split(' ')[0]
-
-        if expected_shasum != actual_shasum:
-            raise ValueError(
-                f'SHA512 mismatch! '
-                f'expected "{expected_shasum}" but got "{actual_shasum}"',
-            )
-
-        return data
+        if code != HTTPStatus.OK:
+            raise ValueError(f'HTTP failure. Code: {code}')
+        return resp.read()
+
+
+def extract(url: str, data: bytes):
+    result = {}
+    if re.search(r'(?i)\.tar\.', url):
+        tmpf = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            with tmpf:
+                tmpf.write(data)
+            with tarfile.open(tmpf.name) as tar:
+                for member in (x for x in tar.getmembers() if x.isfile()):
+                    name = member.name.rpartition('\\')[2].rpartition('/')[2]
+                    result[name] = tar.extractfile(member).read()
+        finally:
+            os.remove(tmpf.name)
+    elif url.lower().endswith('.zip'):
+        with io.BytesIO(data) as bio, zipfile.ZipFile(bio) as zipp:
+            for info in (x for x in zipp.infolist() if not x.is_dir()):
+                name = info.filename.rpartition('\\')[2].rpartition('/')[2]
```

Comment:
this seems odd -- should this use os.path instead?

Reply:
Neither tarfile or zipfile document what they use for path separator, and I believe using os.path does not handle both back and forward slashes. This version should work in all combos.